Makefile

```diff
@@ -0,0 +1,93 @@
+# By default, let's print out some help
+.PHONY: usage
+usage:
+	@echo "$$(tput bold)Welcome to libtock-rs!$$(tput sgr0)"
+	@echo
+	@echo "First things first, if you haven't yet, check out Tocks's doc/Getting_Started."
+	@echo "After that read the README from libtock-rs"
+	@echo "You'll need to install a few requirements before we get going."
+	@echo
+	@echo "The next step is to choose a board to build Tock for. Mainline"
+	@echo "libtock-rs currently includes support for the following platforms:"
+	@for f in `./tools/list_boards.sh`; do printf " - $$f\n"; done
```

Comment:
One last thing: What I think is strange about this line is that you query the list of supported boards from the file system while, in the end, the only thing that matters is whether you explicitly specify the target in the Makefile. If you explicitly enumerate the boards in the Makefile you could do the same here in order to reduce complexity :wink:

Reply:
Fixed